### ISSUE TITLE
`salt-cloud` failing to list nodes for recent versions of `apache-libcloud`

### ISSUE THREAD

**Issue by brianthelion** [saltstack/salt#56589]
### Description of Issue

The `node_state` function in `libcloudfuncs.py` is throwing errors related to the `id_` input type:

```
[DEBUG   ] Failed to execute 'gce.list_nodes()' while querying for running nodes: running
Traceback (most recent call last):
  File "/home/br/Dev/f0cal/2020-04/_venv/lib/python3.6/site-packages/salt/cloud/__init__.py", line 2366, in run_parallel_map_providers_query
    cloud.clouds[data['fun']]()
  File "/home/br/Dev/f0cal/2020-04/_venv/lib/python3.6/site-packages/salt/cloud/libcloudfuncs.py", line 467, in list_nodes
    'state': node_state(node.state)
  File "/home/br/Dev/f0cal/2020-04/_venv/lib/python3.6/site-packages/salt/cloud/libcloudfuncs.py", line 86, in node_state
    return states_str[id_] if isinstance(id_, six.string_types) else states_int[id_]
KeyError: running
```

A quick look at [closed issues](https://github.com/saltstack/salt/issues?q=is%3Aissue+node_state+sort%3Aupdated-desc+) reveals that we've seen this before.

First, `libcloud` was passing a `node.state` as an `int` and then a `str` after v1.0. Now (v2.8.1) it appears to be passing a `NodeState` object that needs to be [coerced](https://github.com/apache/libcloud/blob/3086851e1c7f5838cd9c0fb42a31bdf08b6f9706/libcloud/compute/types.py#L44).

### Setup
N/A

### Steps to Reproduce Issue
Run `salt-cloud` with recent version of `apache-libcloud` and debug on.

### Versions Report

```
Salt Version:
           Salt: 3000.1
 
Dependency Versions:
           cffi: 1.14.0
       cherrypy: Not Installed
       dateutil: 2.8.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
         Jinja2: 2.11.1
        libgit2: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.2
   mysql-python: Not Installed
      pycparser: 2.20
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.6.9 (default, Nov  7 2019, 10:44:02)
   python-gnupg: Not Installed
         PyYAML: 5.3.1
          PyZMQ: 19.0.0
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.2
 
System Versions:
           dist: Ubuntu 18.04 bionic
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-91-generic
         system: Linux
        version: Ubuntu 18.04 bionic
```

**Comment by Akm0d** [comment]
@brianthelion thank you for submitting the issue!  Can you tell me the output of `pip freeze|grep apache-libcloud` and the *exact* salt-cloud command you used to reproduce the bug?  relevant parts of /etc/cloud.profiles.d and /etc/cloud.providers.d may also be helpful :)

**Comment by brianthelion** [comment]
```bash
$ pip freeze | grep apache-libcloud
apache-libcloud==2.8.1
```
The other collateral is slightly more complicated; I'm running this as part of a complex orchestration. Here's what I can give you without too many redactions:

```yaml
# my-orchestration.sls
create_instance:
  salt.runner:
    - name: cloud.profile
    - prof: my-profile
    - provider: my-provider
    - instances:
        - {{ instance_name }}
    - vm_overrides:
        deploy: False
        metadata: '{"enable-oslogin": "FALSE", "ssh-keys": "{{ auth_str }}"}'
```

where:

```yaml
# profile
my-profile:
  provider: my-provider
  image: debian-10
  size: f1-micro
  location: us-central1-a
```

and

```yaml
# provider
my-provider:
  driver: gce
  project: my-project
  service_account_email_address: REDACTED
  service_account_private_key: REDACTED
```

**Comment by s0undt3ch** [comment]
This will be fixed by #59374
